index.js

```diff
@@ -280,6 +281,84 @@ app.get("/label", async (req, res) => {
 
 //#endregion
 
+//#region  --- /user ROUTES
+
+// Register a new user
+app.post('/register', async (req, res) => {
+    try {
+      const {firstname, lastname, username, email, password } = req.body;
+
+      // Hash the password
+      const hashedPassword = await bcrypt.hash(password, 10);
+
+      // Check if user already exists
+      const user = await getUsersByUsername(username);
+      if (user.length !== 0) {
+        return res.status(401).send('Username or email already exists');
+      }
+
+      // Insert the new user into the database
+      const newAccount = await newUser(firstname, lastname, username, email, hashedPassword);
+      if (newAccount.length === 0){
+        res.status(401).json({ error: 'Failed to create account' });
+      } else {
+        res.status(200).send('You have successfully created your account');
+      }
+    } catch (err) {
+        res.status(500).json({ error: err });
+    }
+});
+  
+// Login route
+app.get('/login', async (req, res) => {
```

Comment:
May or may not want to change to a GET with an ID or username that comes after a slash? Don't know what's the best practice here for logins. Maybe username since you do include that in the body in the original API design.

Such as "/users/:id" or "/users/:username"